docs/guide/interactions/views.rst

```diff
@@ -0,0 +1,467 @@
+
+            self.roles: List[discord.Role] = roles
+            for role in roles:
+                self.selector.add_option(label=role.name, value=str(role.id))
+
+        @discord.ui.select(
+            placeholder='Select a role',
+            min_values=1,
+            max_values=1,
+        )
+        async def selector(
+            self,
+            interaction: discord.Interaction,
+            component: discord.ui.Select['RoleSelector'],
+        ) -> None:
+            raise NotImplementedError
+
+We also need to add a body to our ``selector`` callback function, which will assign the selected role to the user.
+This is fairly simple to do, when the callback is invoked we can access the options the user selected via the
+:attr:`~discord.ui.Select.options` attribute. This holds a list of values, but since we limited the number of values to `1`,
+we can just access the first element directly. We can then use sets to determine what roles the user already has, 
+and then override the member's roles with :meth:`Member.edit <discord.Member.edit>`.
+
+Our function body might look like this:
+
+.. code-block:: python
+    :emphasize-lines: 21-25
+
+    class RoleSelector(discord.ui.View):
+        def __init__(self, message_id: int, roles: List[discord.Role]) -> None:
+            super().__init__(timeout=None)
+
+            self.selector.custom_id = f'role_selector:{message_id}'
+
+            self.roles: List[discord.Role] = roles
+            for role in roles:
+                self.selector.add_option(label=role.name, value=str(role.id))
+
+        @discord.ui.select(
+            placeholder='Select a role',
+            min_values=1,
+            max_values=1,
+        )
+        async def selector(
+            self,
+            interaction: discord.Interaction,
+            component: discord.ui.Select['RoleSelector'],
+        ) -> None:
+            role_id = int(component.values[0])
+            role = discord.Object(id=role_id, type=discord.Role)
+            roles = set(interaction.user.roles) - set(self.roles) | {role}
+            await interaction.user.edit(roles=roles)
+            await interaction.response.defer()
+
+
+Making a View Persist
+~~~~~~~~~~~~~~~~~~~~~~
+
+Now our view class is complete, we can manually attach an instance to a message to retrieve the ID.
+When we're doing this, the ``custom_id`` can be set to anything since the view handling will be done 
+when we mark the view as persistent.
+
+.. code-block:: python
+
+    ROLE_IDS = [490320652230852629, 714516281293799438, 859169678966784031, 673195396834656257]
+    roles = [guild.get_role(id) for id in ROLE_IDS]
+    await channel.send('Choose your house', view=RoleSelector(0, roles)) # we don't know the message ID yet.
+
+Once we have a message ID, we just need to create an instance of our view and attach it to the :class:`~discord.Client`
+with the :meth:`~discord.Client.add_view` method.
```

Comment:
```suggestion
Once we have a message ID, we just need to create an instance of our view and attach it to the :class:`~discord.Client`
with the :meth:`~discord.Client.add_view` method. By calling this method in our client setup and using a custom ID on all of the view components, the bot will add this view every time it is started. This is what makes a view persistent.
```
I'm not sure I am perfectly happy with my wording here either, but adding a little more detail describing what exactly this section is for and why this is important for a persistent view seems relevant.

Reply:
Added some extra detail, however I'm not really sure there's much benefit.